code/dirs.py

```diff
@@ -260,3 +307,230 @@ def makedirs_if_not_exist(directory):
+    if os.environ.has_key("APPDATA") and os.environ.has_key("LOCALAPPDATA"):
+        
+        local_appdir = os.environ.has_key("LOCALAPPDATA")
+        roaming_appdir = os.environ.has_key("APPDATA")
+
+        return (roaming_appdir, local_appdir)
+    else:
+        if g.debug: sys.stderr.write("WARNING: APPDATA and LOCALAPPDATA environment not present \n")
+
+    # Ok, we could try to get home user directories.
+    # Get it to all different way.
+    home_dirs = []
+
+    if 'ctypes' in sys.modules:
+        for func_name in win32_function:
+            try:
+                func = ctypes.windll.shell32.getattr(func_name)
+                
+                handler = globals().get("shell32_%s" % func_name)
+                
+                profiledir = handler(func, 'home')
+                
+                if profiledir is not None:
+                    home_dirs.append(profiledir)
+                else:
+                    if g.debug: sys.stderr.write("WARNING: '%s' shell32 function fails for home\n" % func_name)
+            except:
+                if g.debug: sys.stderr.write("WARNING: Impossible to use '%s' shell32 function for home\n" % func_name)
+
+    if os.environ.has_key("USERPROFILE")
+        home_dirs.append(os.environ.get("USERPROFILE"))
+    else:
+        if g.debug: sys.stderr.write("WARNING: USERPROFILE environment not present \n")
+
+    if os.environ.has_key("HOMEPATH") and os.environ.has_key("HOMEDRIVE")
+        home_dirs.append(os.environ.get("HOMEDRIVE") + os.environ.get("HOMEPATH"))
+    else:
+        if g.debug: sys.stderr.write("WARNING: HOMEPATH and HOMEDRIVE environment not present \n")
+
+    if os.environ.has_key("USERNAME")
+        home_dirs.append("C:\\Users\\" + os.environ.get("USERNAME"))
+        home_dirs.append("C:\\Documents and Settings\\" + os.environ.get("USERNAME"))
+    else:
+        if g.debug: sys.stderr.write("WARNING: USERNAME environment not present \n")
+
+    # Now we can check if we find AppData.
+    for home in home_dirs:
+        local_appdir = os.path.join(home, "AppData", "Local")
+        roaming_appdir = os.path.join(home, "AppData", "Roaming")
+    
+        if (os.path.is_dir(local_appdir) and os.path.is_dir(roaming_appdir))
+            return (roaming_appdir, local_appdir)
+
+    # We can check the old ApplicationData
+    for home in home_dirs:
+        appdir = os.path.join(home, "Application Data")
+    
+        if (os.path.is_dir(appdir))
+            return (appdir, appdir)
+
+    # Ok, let just use the user directory.
+    for home in home_dirs:
+        if (os.path.is_dir(home))
+            return (home, home)
+
+    # Now, next we could be check the registry.
+    # But it's not good idea.
+    # 1) Registry shell32 are considered as unreliable
+    # 2) If we can't use win32 for shell32, why could we use it for registry ?
+
+    return False
+
+def get_mac_library():
+    
+    try:
+        import ctypes
+        import ctypes.util
+    except:
```

Comment:
Please use `except ImportError:`.  We should definitely not catch every throwable here (we do not want to handle requests to shutdown or out or memory errors).